ethereum/executionlayer/src/main/java/tech/pegasys/teku/ethereum/executionlayer/ExecutionBuilderModule.java

```diff
@@ -404,26 +404,44 @@ private void logReceivedBuilderExecutionPayload(final ExecutionPayload execution
   private void logReceivedBuilderBid(final BuilderBid builderBid) {
     final ExecutionPayloadHeader payloadHeader = builderBid.getExecutionPayloadHeader();
     LOG.info(
-        "Received Builder Bid (Block Number = {}, Block Hash = {}, MEV Reward (wei) = {}, Gas Limit = {}, Gas Used = {})",
+        "Received Builder Bid (Block Number = {}, Block Hash = {}, MEV Reward (ETH) = {}, Gas Limit = {}, Gas Used = {})",
         payloadHeader.getBlockNumber(),
         payloadHeader.getBlockHash(),
-        builderBid.getValue().toDecimalString(),
+        weiToEth(builderBid.getValue()),
         payloadHeader.getGasLimit(),
         payloadHeader.getGasUsed());
   }
 
   private void logLocalPayloadWin(final UInt256 builderBidValue, final UInt256 localPayloadValue) {
     if (builderBidCompareFactor.isEmpty() || builderBidCompareFactor.get() == HUNDRED_PERCENT) {
       LOG.info(
-          "Local execution payload ({} Gwei) is chosen over builder bid ({} Gwei).",
-          localPayloadValue.divide(WEI_TO_GWEI).toDecimalString(),
-          builderBidValue.divide(WEI_TO_GWEI).toDecimalString());
+          "Local execution payload ({} ETH) is chosen over builder bid ({} ETH).",
+          weiToEth(localPayloadValue),
+          weiToEth(builderBidValue));
     } else {
       LOG.info(
-          "Local execution payload ({} Gwei) is chosen over builder bid ({} Gwei, compare factor {}%).",
-          localPayloadValue.divide(WEI_TO_GWEI).toDecimalString(),
-          builderBidValue.divide(WEI_TO_GWEI).toDecimalString(),
-          builderBidCompareFactor.get());
+          "Local execution payload ({} ETH) is chosen over builder bid ({} ETH, compare factor {}%).",
+          weiToEth(localPayloadValue), weiToEth(builderBidValue), builderBidCompareFactor.get());
     }
   }
+
+  private String weiToEth(final UInt256 wei) {
+    final String weiString = wei.toDecimalString();
+    final StringBuilder stringBuilder = new StringBuilder(20);
+
+    if (weiString.length() > 18) {
+      final int pointPosition = weiString.length() - 18;
+      final String ethInteger = weiString.substring(0, pointPosition);
+      final String ethDecimals = weiString.substring(pointPosition);
+
+      stringBuilder.append(ethInteger).append(".").append(ethDecimals);
+    } else {
+      final int paddingZeros = 18 - weiString.length();
+      stringBuilder.append("0.");
+      IntStream.range(0, paddingZeros).forEach(__ -> stringBuilder.append("0"));
+      stringBuilder.append(weiString);
+    }
+
+    return stringBuilder.toString();
+  }
```

Comment:
so we'll want to think about this...

 - we want it to be fast, and we want tests.... Not sure whether we should be somehow doing this mathematically, or some kind of hybrid approach... but unit tests would let us see the ins and outs...

Reply:
Maybe we should consider using something like https://github.com/web3j/web3j/blob/master/utils/src/main/java/org/web3j/utils/Convert.java ?

Reply:
From their tests it looks like it does exactly what we want: https://github.com/web3j/web3j/blob/master/utils/src/test/java/org/web3j/utils/ConvertTest.java#L38-L40